lib/eslint.js

```diff
@@ -718,11 +720,20 @@ module.exports = (function() {
 
                 severity = getRuleSeverity(config.rules[key]);
                 options = getRuleOptions(config.rules[key]);
+                parserName = config.parser;
+
+                // for use against pre-parsed AST, may be no valid parser configured
+                try {
+                    parser = require(parserName);
```

Comment:
I tried to match the same loading strategy; looks like it's loaded once per execution of `parse`--which makes sense given that the config may be different depending on the file path. Same is true for `RuleContext`, as I understand it.

I would be willing to pass `parse` bound to config, i.e.
```js
parser = function (text) { return parse(text, config); };
```
to keep it all flowing through the same place, but it was unclear what the impact on `messages` would be if a parse error is encountered while parsing some other JS (in my case, some dependency). 

In my own human static analysis, it seems by the time the code has already flowed through `parse` once and was able to `require` the configured parser, or else `RuleContext` would not be instantiated (https://github.com/benmosher/eslint/blob/parser/lib/eslint.js#L692) as there would be no AST. As such, the swallowed exception might only be encountered when providing an external AST, at which point a null `parser` seems to correctly convey that no parser is configured.

Also, you had mentioned wanting access to [the whole module](https://github.com/eslint/eslint/issues/3670#issuecomment-140490898) within a plugin; not necessary for my purposes--I just need an AST from a full module--but I can understand wanting to keep it open for others.